FirebaseSessions/Sources/Public/SessionsSubscriber.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Sessions Subscriber is an interface that dependent SDKs
+/// must implement.
+@objc(FIRSessionsSubscriber)
+public protocol SessionsSubscriber {
+  func onSessionChanged(_ session: SessionPayload)
+  var isDataCollectionEnabled: Bool { get }
+  var sessionsSubscriberName: SessionsSubscriberName { get }
+}
+
+/// Session Payload is a container for Session Data passed to Subscribers
+/// whenever the Session changes
+@objc(FIRSessionPayload)
+public class SessionPayload: NSObject {
```

Comment:
NIT: Can we rename this to "SessionDetails" or "SessionInfo" - Payload sounds more like data to upload/download over network.

Reply:
I like SessionDetails

Reply:
Ack!